### PR TITLE
Split setting distribution information out from TileAndDistributeToWorkgroupsPass

### PR DESCRIPTION
**Pull request by hanhanW**
The distribution information is known when tiling sizes are set. It can
be a separate pass, and be used before different distribution mechanism.
E.g., this is needed to enable distribution in LinalgFuseExpert.

### CODE REVIEW

File: iree/compiler/Codegen/Common/test/tile_and_distribute_to_workgroups.mlir [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// RUN: iree-opt -pass-pipeline='hal.executable(hal.executable.variant(builtin.module(func.func(iree-codegen-tile-and-distribute-to-workgroups)))), canonicalize, cse' -split-input-file %s | FileCheck %s
```
Comment (MaheshRavishankar) [inline iree/compiler/Codegen/Common/test/tile_and_distribute_to_workgroups.mlir:1]: Would you mind just splitting this test into two parts then. Basically the result of `iree-codegen-insert-distribution-info` should become the input MLIR for `iree-codegen-tile-and-distribute-to-workgroups`.

File: iree/compiler/Codegen/Passes.h [hunk 2]
```diff
@@ -119,6 +119,8 @@ std::unique_ptr<OperationPass<FuncOp>> createVectorizePadPass();
 /// Pass to optimize vector transfer_read and transfer_write.
 std::unique_ptr<OperationPass<FuncOp>> createOptimizeVectorTransferPass();
 
+std::unique_ptr<OperationPass<FuncOp>> createInsertDistributionInfoPass();
```
Comment (MaheshRavishankar) [inline iree/compiler/Codegen/Passes.h:122]: Add some comments about this ?

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
Overall looks good. Mostly one question about maybe splitting the lit test?

File: iree/compiler/Codegen/Common/test/tile_and_distribute_to_workgroups.mlir (hunk 1 above)
Comment (hanhanW) [reply iree/compiler/Codegen/Common/test/tile_and_distribute_to_workgroups.mlir]: I think we don't have to change the input MLIR for `iree-codegen-tile-and-distribute-to-workgroups`. We can test the result IR and remove the check about `hal.executable.entry_point`.

**Review by MaheshRavishankar — APPROVED** [review]
Nice! Thanks!

**Pull request merged**